src/DataDictionary/DataDictionary.jsx

```diff
@@ -90,9 +90,10 @@ export function category2NodeList(dictionary) {
   )
     .reduce(
       (lookup, node) => {
-        if (!lookup[node.category]) { lookup[node.category] = []; }
-        lookup[node.category].push(node);
-        return lookup;
+        let category = lookup[node.category];
+        if (!lookup[node.category]) { category = []; }
+        category.push(node);
+        return Object.assign({}, lookup, { [node.category]: category });
```

Comment:
Why are we doing this?
Every time through the reduce loop we're making a new copy of the accumulator dictionary - so we go from O(n) to O(n^2) ... there's no reason to do that

Reply:
Linting gives the error `Assignment to property of function parameter 'lookup'  no-param-reassign` so I am trying to eliminate that, I can disable this for `.reduce` though?

Reply:
Yeah - that would be great if you can eliminate that rule for .reduce :+1: